File: solutions/q1071/q1071_solution.cs
public class Solution {
    public string GcdOfStrings(string str1, string str2) {
        // verifies that str1 + str2 are pallindrome-esque
        if (!(str1 + str2).Equals(str2 + str1)) {
            return "";
        }
        
        /*
            While loop condition: strings are not equal

            If statement condition: if str1 is longer than str2, shrink str1 by the length of str2
            Else condition: if str2 is longer than str1, shrink str2 by the length of str1
            When str1 and str2 become equal, the loop will exit
        */
        while (!str1.Equals(str2)) {
            if (str1.Length > str2.Length) {
                str1 = str1.Substring(str2.Length);
            } else {
                str2 = str2.Substring(str1.Length);
            }
        }
        
        return str1;
    }
}

File: solutions/q1071/q1071_solution.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        if (str1+str2).__eq__(str2+str1) == False:
            return ""

        while str1.__eq__(str2) == False:
            if len(str1) > len(str2):
                str1 = str1[len(str2):]
            else:
                str2 = str2[len(str1):]
        
        return str1
